src/core/internal/arrayop.d

```diff
@@ -302,6 +362,25 @@ template not(alias tmlp)
 {
     enum not(Args...) = !tmlp!Args;
 }
+/// find index for which pred!haystack[index] is true or return -1
```

Comment:
Please use Ddoc style comments, with `Params:` and `Returns:` sections. It's not just for readability, it's so tools can be used on the documentation.